tests/foreman/maintain/test_backup_restore.py

```diff
@@ -409,6 +408,89 @@ def test_negative_restore_baddir(sat_maintain, setup_backup_tests):
     assert BADDIR_MSG in str(result.stdout)
 
 
+@pytest.mark.parametrize('backup_type', ['online', 'offline'])
+def test_positive_puppet_backup_restore(
+    sat_maintain,
+    setup_backup_tests,
+    module_synced_repos,
+    backup_type,
+):
+    """Puppet backup/restore test.
+
+    :id: 0f8555dc-8c1f-47cd-b5f6-5655d98564cd
+
+    :parametrized: yes
+
+    :steps:
+        1. Enable puppet on the satellite
+        2. create a backup of different types
+        3. check that appropriate files are created
+        4. restore the backup (installer --reset-data is run in this step)
+        5. check system health
+        6. check the content was restored
+
+    :expectedresults:
+        1. backup succeeds
+        2. expected files are present in the backup
+        3. restore succeeds
+        4. system health check succeeds
+        5. content is present after restore
+
+    :customerscenario: true
+
+    :BZ: 2158896
+    """
+    # Enable puppet on the Satellite
+    sat_maintain.enable_puppet_satellite()
```

Comment:
@Griffin-Sullivan Can we not use session_puppet_enabled_sat here instead or else disable Puppet in the teardown?

Reply:
I guess we could, but I felt that it was safer keeping sat_maintain since that is what the test suite was built around and I didn't want to mess with the test. Although, if we make this change I will now need to update master, so I would rather merge this and if you feel it's necessary to use the fixture then I will add a PR updating master and then cherry pick that to 6.13.z.

Reply:
Sounds good, using that fixture could save time and wouldn't conflict with the sat_maintain fixture so other tests could run without errors if the installer fails, and also you could handle disabling Puppet in the teardown as well to workaround